estate/models/estate_property.py

```diff
@@ -0,0 +1,102 @@
+from datetime import date
+
+from dateutil.relativedelta import relativedelta
+
+from odoo import api, fields, models
+from odoo.exceptions import UserError
+from odoo.tools import float_compare
+
+
+class EstateProperty(models.Model):
+    _name = "estate.property"
+    _description = "Real Estate Property"
+    _order = "id desc"
+
+    name = fields.Char(string="Title", required=True)
+    property_type = fields.Selection(
+        string='Property Type',
+        selection=[('house', 'House'), ('apartment', 'Apartment')])
+    description = fields.Text()
+    postcode = fields.Char()
+    date_availability = fields.Date(copy=False, readonly=True, default=lambda self: date.today() + relativedelta(months=3))
+    expected_price = fields.Float("Expected Price", required=True)
+    selling_price = fields.Float("Selling Price", readonly=True, copy=False)
+    bedrooms = fields.Integer(default=2)
+    living_area = fields.Integer(string='Living Area (m2)')
+    facades = fields.Integer()
+    garage = fields.Boolean()
+    garden = fields.Boolean()
+    garden_area = fields.Integer(string='Garden Area (m2)')
+    garden_orientation = fields.Selection(
+        string='Garden Orientation',
+        selection=[('north', 'North'), ('south', 'South'),
+        ('east', 'East'), ('west', 'West')])
+    active = fields.Boolean(default=True)
+    state = fields.Selection(
+    copy=False,
+    readonly=True,
+    default='new',
+    string='States',
+    selection=[
+        ('new', 'New'),
+        ('offer_received', 'Offer Received'),
+        ('offer_accepted', 'Offer Accepted'),
+        ('sold', 'Sold'),
+        ('cancelled', 'Cancelled')]
+    )
+
+    def action_set_sold(self):
+        if self.state != "cancelled":
+            self.state = "sold"
+        else:
+            raise UserError("A cancelled property can not be sold")
+        return True
+
+    def action_set_cancelled(self):
+        if (self.state != "sold"):
+            self.state = "cancelled"
+        else:
+            raise UserError("A sold property can not be cancelled")
+        return True
+
+    property_type_id = fields.Many2one('estate.property.type', string='Type')
+    buyer_id = fields.Many2one('res.partner', string='Buyer', copy=False)
+    salesman_id = fields.Many2one('res.users', string='Salesman', default=lambda self: self.env.user)
+    tags_ids = fields.Many2many('estate.property.tags', string='Tags')
+    offer_ids = fields.One2many("estate.property.offer", "property_id", string="offer")
+    total_area = fields.Integer(string='Total Area(m2)', compute='_compute_total_area', store=True)
+    best_price = fields.Float(string='Best Offer', compute='_compute_best_price', store=True)
+
+    @api.depends('living_area', 'garden_area')
+    def _compute_total_area(self):
+        for property in self:
+            property.total_area = property.living_area + (property.garden_area or 0)
```

Comment:
```suggestion
            property.total_area = property.living_area + property.garden_area
```
garden_area is an integer, it cannot be False